### PR TITLE
rust-analyzer: update to 2021.07.05

### PR DESCRIPTION
**Pull request by Logarithmus**
#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me

### CODE REVIEW

**Comment by ericonr** [comment]
Pull in changes from https://github.com/void-linux/void-packages/pull/31085

**Comment by Logarithmus** [comment]
> Pull in changes from #31085

I've replaced `do_check()` with `make_check=no`, I think that's all?

**Comment by ericonr** [comment]
And keep the comment.

File: srcpkgs/rust-analyzer/template [hunk 1]
```diff
@@ -11,14 +11,11 @@ maintainer="Gabriel Sanches <gabriel@gsr.dev>"
...
 homepage="https://rust-analyzer.github.io/"
 distfiles="https://github.com/${pkgname}/${pkgname}/archive/${_ver}.tar.gz"
-checksum=75d810a4b3fd08d8d0948186f077b754ab9a2a02af3d3c6947e0baa58576e43d
+checksum=a281bab5e119134fc563b37acc9a70bc14b611cff4ae0574650d5802b2716ccf
+make_check=no
 
 export RUST_ANALYZER_REV=${_ver}
 
-do_check() {
-	: # tests require Rust source code
```
Comment (ericonr) [inline srcpkgs/rust-analyzer/template:19]: This comment is relevant.

Comment (Logarithmus) [reply srcpkgs/rust-analyzer/template:19]: Fixed

**Pull request merged**